Loadable.svelte

```diff
@@ -127,15 +130,18 @@
     state = STATES.SUCCESS
     component = LOADED.get(loader)
   } else {
-    onMount(async () => {
-      await load()
-      dispatch('load')
-      if (unloader) {
-        return () => {
-          LOADED.delete(loader)
-          if (typeof unloader === 'function') {
-            unloader()
-          }
+    onMount(() => {
+      mounted = true;
+      load().then(() => {
+        if (mounted) {
+          dispatch('load')
+        }
+      })
+
+      return () => {
+        mounted = false;
+        if (typeof unloader === 'function') {
+          unloader()
         }
       }
     })
```

Comment:
A possible alternative would be to wait for the load promise to finish to remove the component from the cache

```suggestion
    onMount(() => {
      const loadPromise = load()

      loadPromise.then(() => {
        if (mounted) {
          dispatch('load')
        }
      })

      return () => {
        mounted = false

        loadPromise.then(() => {
          LOADED.delete(loader)
        })

        if (!unloader) return
        if (typeof unloader === 'function') {
          unloader()
        }
      }
    })
```

Reply:
Or we could add a second parameter `cache` to the `load` and replace the 

```
      component = await loadComponent(loader)
```

with

```
      component = await loadComponent(loader, false)
      if (mounted) {
        LOADED.set(loader, component)
      }
```

Reply:
with 

```
  loadPromise.then(() => {
    LOADED.delete(loader)
  })
```

that would be a separate race condition. The unmount code could run much later - perhaps after this same Loadable re-mounts, potentially doing undesirable things. 


```
 component = await loadComponent(loader, false)
      if (mounted) {
        LOADED.set(loader, component)
      }
```

that would definitely work. Do you want me to change it to that? I don't see why that's any better than just declining to cache it in the first place, if you know you don't want it cached.

Reply:
I also wonder if bad things can happen with the 

```
component = await loadComponent(loader, false)
      if (mounted) {
        LOADED.set(loader, component)
      }
```

approach if loader reactively changes. Very obscure use case, possibly not even supported. Regardless though, simply not caching the loader when not desired seems preferable to trying to rig the asynchrony so it can be properly removed.

Is there any reason why you want to have the loader temporarily cached when unloader is true? This isn't React - we don't need to worry about a ton of re-renders firing, not finding the cached loader, and then re-running it over and over.

Reply:
I'm only having a bit of trouble accepting the semantics of passing the `unloader` prop to the static load method. I was thinking of a scenario where you could have more than one loadable component loading and rendering the same `loader` at some given point in time. If we pass an `unloader` prop to them, they won't be able to use the cached value. That's why I was thinking of removing it from the cache only `onDestroy`. Maybe I'm over-thinking and this case is rare/doesn't make sense or the browser cache would be enough.

Reply:
@kaisermann I do think you're over-thinking it. In this really care case, your bundler would already have the module loaded and cached (either in webpack or Rollup), or if you're using native ESM, then it would definitely be cached. And since you have that nice delay property, re-firing that resolved promise each time would not even result in a flash of the loading spinner.

Reply:
@kaisermann but even here, if you want nice shared caching like this ... just don't pass an unloader prop. Your component already caches by default, and it works. These fixes just fix the functionality of *turning off* said caching (ie, if perhaps you want your loader function to also load data, alongside your component). You might want that to run every time you browse to /contacts (or whatever). Always load your current /contacts data, and also load the contacts component (which will be cached on the second and all future loads, by either webpack or rollup, or by the browsers native ESM functionality)

Reply:
@kaisermann how are we looking on this? Is there anything else you're not sure about on this PR, or does it look good to merge?